src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx

```diff
@@ -42,42 +43,49 @@ export const useIsSnapInstalled = (): {
   isSnapInstalled: boolean | null
   isCorrectVersion: boolean | null
 } => {
-  const [isSnapInstalled, setIsSnapInstalled] = useState<null | boolean>(null)
-  const [isCorrectVersion, setIsCorrectVersion] = useState<boolean | null>(null)
-
   const {
     state: { wallet, isConnected },
   } = useWallet()
 
   const connectedRdns = useAppSelector(selectWalletRdns)
 
   const checkSnapInstallation = useCallback(async () => {
-    if (connectedRdns !== METAMASK_RDNS) return setIsSnapInstalled(null)
-    if (!isConnected) return setIsSnapInstalled(null)
+    if (connectedRdns !== METAMASK_RDNS || !isConnected)
+      return {
+        isCorrectVersion: null,
+        isSnapInstalled: null,
+      }
+
     const isMetaMaskDesktop = checkIsMetaMaskDesktop(wallet)
-    if (!isMetaMaskDesktop) return
+    if (!isMetaMaskDesktop)
+      return {
+        isCorrectVersion: null,
+        isSnapInstalled: null,
+      }
 
     const version = await getSnapVersion()
     const _isSnapInstalled = await checkIsSnapInstalled()
 
-    setIsCorrectVersion(version === snapVersion)
-    setIsSnapInstalled(_isSnapInstalled)
+    return {
+      isCorrectVersion: version === snapVersion,
+      isSnapInstalled: _isSnapInstalled,
+    }
   }, [connectedRdns, isConnected, wallet])
 
-  useEffect(() => {
-    // Call the function immediately
-    checkSnapInstallation()
-
-    // Set up a polling interval
-    const intervalId = setInterval(checkSnapInstallation, POLL_INTERVAL)
-
-    // Clean up the interval when the component unmounts
-    return () => clearInterval(intervalId)
-  }, [checkSnapInstallation, wallet])
+  const { data: snapInstallation = { isCorrectVersion: null, isSnapInstalled: null } } = useQuery({
+    queryKey: ['snapInstallation', connectedRdns, isConnected],
+    queryFn: checkSnapInstallation,
+    refetchInterval: POLL_INTERVAL,
+    staleTime: 5 * 60 * 1000, // 5 minutes
+    gcTime: 5 * 60 * 1000, // 5 minutes
+  })
 
   const data = useMemo(
-    () => ({ isSnapInstalled, isCorrectVersion }),
-    [isSnapInstalled, isCorrectVersion],
+    () => ({
+      isSnapInstalled: snapInstallation.isSnapInstalled,
+      isCorrectVersion: snapInstallation.isCorrectVersion,
+    }),
+    [snapInstallation],
```

Comment:
We don't need that anymore, we can consume snapInstallation directly

Reply:
Ohh indeed 🧠  the default assignment on `data` above makes it so we can simply return it as-is, since `data` will always be defined. Nice catch ser! 9ea5e6c403